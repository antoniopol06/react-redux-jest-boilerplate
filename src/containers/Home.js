import React from 'react';
import SearchBox from './../components/SearchBox';
import RelatedWords from './../components/RelatedWords';
import ReduxComponent from './ReduxComponent';
import {fetchRelatedWords} from './../actions';

class Home extends ReduxComponent {
    search = (value) => {
        if (value !== this.status().selectedWord) {
            this.dispatch(fetchRelatedWords(value));
        }
    };

    render() {
        let loading = !!this.status().fetching;

        return (
            <div>
                <SearchBox placeholder="Search..." submit={this.search} disabled={loading}/>
                {loading ? (
                    <p><em>Loading...</em></p>
                ) : (
                    this.status().selectedWord ? (
                        <RelatedWords
                            selected={this.status().selectedWord}
                            words={this.status().relatedWords}
                        />
                    ) : ''
                )}
            </div>
        );
    }
}

export default Home;
