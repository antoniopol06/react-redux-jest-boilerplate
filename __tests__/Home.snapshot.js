import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import mockStore from 'redux-mock-store';
import Home from './../src/containers/Home';

test('Home snapshot', () => {
    const store = mockStore({});

    const container = renderer.create(
        <Provider store={store}>
            <Home/>
        </Provider>
    );

    let tree = container.toJSON();
    expect(tree).toMatchSnapshot();
});
