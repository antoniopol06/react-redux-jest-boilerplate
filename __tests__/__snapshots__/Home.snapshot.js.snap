exports[`test Home snapshot 1`] = `
<div>
  <div
    className="App-search-box">
    <input
      className="App-search-box-input"
      onKeyPress={[Function]}
      placeholder="Search..."
      readOnly={false}
      type="text" />
    <button
      className="App-search-box-btn"
      disabled={false}
      onClick={[Function]}>
      Search
    </button>
  </div>
  
</div>
`;
